File: api/src/test/java/net/blueshell/api/testsupport/DoNotIncludeFactory.java
package net.blueshell.api.testsupport;

import com.tngtech.archunit.core.importer.ImportOption;
import com.tngtech.archunit.core.importer.Location;

/**
 * ArchUnit import option excluding factories from the application scan.
 */
public class DoNotIncludeFactory implements ImportOption {
    @Override
    public boolean includes(Location location) {
        return !location.contains("factory");
    }
}

File: api/src/test/kotlin/net/blueshell/api/testsupport/DoNotIncludeFactory.kt
package net.blueshell.api.testsupport

import com.tngtech.archunit.core.importer.ImportOption
import com.tngtech.archunit.core.importer.Location

/**
 * ArchUnit import option excluding factories from the application scan.
 */
class DoNotIncludeFactory : ImportOption {
    override fun includes(location: Location): Boolean =
        !location.contains("factory")
}
